src/routes/teams.rs

```diff
@@ -1,25 +1,227 @@
+        } else {
+            Err(ApiError::AuthenticationError)
+        }
+    } else {
+        Err(ApiError::AuthenticationError)
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct EditTeamMember {
+    pub permissions: Option<Permissions>,
+    pub role: Option<String>,
+}
+
+#[patch("{id}/members/{user_id}")]
+pub async fn edit_team_member(
+    req: HttpRequest,
+    info: web::Path<(TeamId, UserId)>,
+    pool: web::Data<PgPool>,
+    edit_member: web::Json<EditTeamMember>,
+) -> Result<HttpResponse, ApiError> {
+    let ids = info.into_inner();
+    let id = ids.0.into();
+    let user_id = ids.1.into();
+
+    let current_user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+    let team_member = TeamMember::get_from_user_id(id, current_user.id.into(), &**pool).await?;
+
+    if let Some(member) = team_member {
+        if member.permissions.contains(Permissions::EDIT_MEMBER)
+            && edit_member.role.as_deref() != Some(crate::models::teams::OWNER_ROLE)
+        {
+            if let Some(new_permissions) = edit_member.permissions {
+                if !member.permissions.contains(new_permissions) {
+                    return Err(ApiError::AuthenticationError);
+                }
+            }
+
+            TeamMember::edit_team_member(
+                id,
+                user_id,
+                edit_member.permissions.map(|x| x.bits() as i64),
+                edit_member.role.clone(),
+                None,
+                &**pool,
+            )
+            .await?;
+
+            Ok(HttpResponse::Ok().body(""))
+        } else {
+            Err(ApiError::AuthenticationError)
+        }
+    } else {
+        Err(ApiError::AuthenticationError)
+    }
+}
+
+#[delete("{id}/members/{user_id}")]
+pub async fn remove_team_member(
+    req: HttpRequest,
+    info: web::Path<(TeamId, UserId)>,
+    pool: web::Data<PgPool>,
+) -> Result<HttpResponse, ApiError> {
+    let ids = info.into_inner();
+    let id = ids.0.into();
+    let user_id = ids.1.into();
+
+    let current_user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+    let team_member = TeamMember::get_from_user_id(id, current_user.id.into(), &**pool).await?;
+
+    if let Some(member) = team_member {
+        let delete_member_option = TeamMember::get_from_user_id(id, user_id, &**pool).await?;
+
+        if let Some(delete_member) = delete_member_option {
+            if delete_member.accepted {
+                if member.permissions.contains(Permissions::DELETE_MOD) {
+                    TeamMember::delete(id, user_id, &**pool).await?;
+                    Ok(HttpResponse::Ok().body(""))
+                } else {
+                    Err(ApiError::AuthenticationError)
+                }
+            } else {
+                if member.permissions.contains(Permissions::MANAGE_INVITES) {
```

Comment:
This should allow users to reject team member requests; if the user is the same as the user for the team member being deleted, this shouldn't require any permissions.